File: TestSample/go/person/person.go
package person

import "fmt"

type Person interface {
	Speak() string
	Hello() string
	Name() string
	Language() string
}

type Base struct {
	name     string
	language string
}

func NewBase(name, language string) Base {
	return Base{name: name, language: language}
}

func (b Base) Name() string     { return b.name }
func (b Base) Language() string { return b.language }

func Format(p Person) string {
	return fmt.Sprintf("[%s] %s: %s", p.Language(), p.Name(), p.Speak())
}

File: TestSample/java/src/main/java/helloworld/Person.java
package helloworld;

public abstract class Person {
    protected final String name;
    protected final String language;

    protected Person(String name, String language) {
        this.name = name;
        this.language = language;
    }

    public abstract String speak();

    public String hello() {
        return String.format("[%s] %s: %s", language, name, speak());
    }
}
